File: donggyun/greedy/조이스틱.js
function solution(name) {
  let answer = 0;
  let min_move = name.length - 1;

  [...name].map((n, i) => {
    answer += Math.min(n.charCodeAt() - 65, 91 - n.charCodeAt());
    let idx = i + 1;

    // 연속되는 A의 개수
    while (idx < name.length && name[idx] === "A") {
      idx++;
    }

    min_move = Math.min(
      min_move, // 순차적으로 이동
      i * 2 + name.length - idx, // 오른쪽으로 갔다가 다시 돌아서 가는 경우
      i + 2 * (name.length - idx) // 왼쪽으로 갔다가 다시 돌아오는 경우
    );
  });

  return answer + min_move;
}

File: Grace/greedy/조이스틱.py
def solution(name): 
    answer = 0
    # 상, 하
    tb = 0
    # 좌, 우는 사실상 name의 길이 - 1
    lr = len(name) - 1
    
    # A를 고려하기 위함.
    next = 0
    
    # (0, 'J') (1, 'E') (2, 'R') ... 
    for i, char in enumerate(name):
        # 상, 하 중 더 차이가 적은 곳
        tb += min(ord(char) - ord('A'), ord('Z') - ord(char) + 1) 
        
        # A 탐색
        next = i + 1
        
        while next < len(name) and name[next] == 'A':
            next += 1
        
        # 기존, A를 왼쪽으로 검사, A를 오른쪽으로 검사 중 최소를 찾음.
        lr = min([lr, i * 2 + len(name) - next, i + 2 * (len(name) - next)])
        
    # 좌우 와 상하 를 더함
    answer = lr + tb
    return answer